Wikipedia/Code/MWKImage.m

```diff
@@ -272,90 +218,34 @@ - (BOOL)isEqual:(id)object {
 }
 
 - (BOOL)isEqualToImage:(MWKImage*)image {
-    return self == image || [self.fileName isEqualToString:image.fileName];
+    return self == image || [[self.sourceURLString wmf_schemelessURL] isEqualToString:[image.sourceURLString wmf_schemelessURL]];
 }
 
 - (NSUInteger)hash {
     return [self.fileName hash];
 }
 
+- (BOOL)isPathEqualToImage:(MWKImage*)image {
+    return [[[self.sourceURLString wmf_schemelessURL] stringByDeletingLastPathComponent] isEqualToString:[[image.sourceURLString wmf_schemelessURL] stringByDeletingLastPathComponent]];
+}
+
 - (BOOL)isVariantOfImage:(MWKImage*)otherImage {
     // !!!: this might not be reliable due to underscore, percent encodings, and other unknowns w/ image filenames
     return [self.fileNameNoSizePrefix isEqualToString:otherImage.fileNameNoSizePrefix]
-           && ![self isEqualToImage:otherImage];
+           && [self isPathEqualToImage:otherImage];
```

Comment:
@joewalsh yes i think so - see the test here:
https://github.com/wikimedia/wikipedia-ios/pull/747/commits/c1da552df3dbdd851ab70ca8f25be8b4019144ef#diff-7dc0541c8d97acb98df6dc9574971ca6R20

It turns out the image scaler can use a format image name like "XXXpx-thumbnail.jpg" if the image name is very long - you can see this in the urls used for the test linked above.